src/core/ext/client_channel/parse_address.c

```diff
@@ -119,9 +120,29 @@ int parse_ipv6(grpc_uri *uri, grpc_resolved_address *resolved_addr) {
   memset(in6, 0, sizeof(*in6));
   resolved_addr->len = sizeof(*in6);
   in6->sin6_family = AF_INET6;
-  if (inet_pton(AF_INET6, host, &in6->sin6_addr) == 0) {
-    gpr_log(GPR_ERROR, "invalid ipv6 address: '%s'", host);
-    goto done;
+
+  char *host_end = (char *)gpr_memrchr(host, '%', strlen(host));
```

Comment:
I assume that grpc_uri_parse() has already handled the %-encoding, so that "%25" will have been turned into "%" by the time we get here?